server/handlers.go

```diff
@@ -356,19 +373,25 @@ func (s *Server) handleConnectorCallback(w http.ResponseWriter, r *http.Request)
 	var identity connector.Identity
 	switch conn := conn.Connector.(type) {
 	case connector.CallbackConnector:
-		if r.Method != "GET" {
+		if r.Method != http.MethodGet {
 			s.logger.Errorf("SAML request mapped to OAuth2 connector")
 			s.renderError(w, http.StatusBadRequest, "Invalid request")
 			return
 		}
 		identity, err = conn.HandleCallback(parseScopes(authReq.Scopes), r)
 	case connector.SAMLConnector:
-		if r.Method != "POST" {
-			s.logger.Errorf("OAuth2 request mapped to SAML connector")
+		var response string
+		switch r.Method {
+		case http.MethodGet:
+			response = r.URL.Query().Get("SAMLResponse")
+		case http.MethodPost:
+			response = r.PostFormValue("SAMLResponse")
+		default:
+			s.logger.Errorf("Invalid method responds to SAML callback")
 			s.renderError(w, http.StatusBadRequest, "Invalid request")
 			return
 		}
-		identity, err = conn.HandlePOST(parseScopes(authReq.Scopes), r.PostFormValue("SAMLResponse"), authReq.ID)
+		identity, err = conn.HandleResponse(parseScopes(authReq.Scopes), response, authReq.ID)
```

Comment:
So in this case I could request `responseBinding: post`, get an HTTP redirect response, and we'd still process it even though we've explicitly stated that we don't.

I think we need to enforce that only the binding that the connector is configured with is actually used.

Reply:
You are right. I will fix this.

Reply:
The problem is at this layer, it doesn't have the knowledge about bindings. Anyway, let me figure it out...